test/cpp/api/autograd.cpp

```diff
@@ -196,6 +196,33 @@ TEST(CustomAutogradTest, FunctionReturnsInput) {
   ASSERT_VARIABLE_EQ(x.grad(), torch::full(1, 2.));
 }
 
+TEST(CustomAutogradTest, FunctionReturnsUndefined) {
+  struct MyFunction : public Function<MyFunction> {
+    static Variable forward(AutogradContext *ctx, Variable var) {
+      return var * 2;
+    }
+
+    static variable_list backward(AutogradContext *ctx, variable_list grad_output) {
+      at::Tensor undefined_tensor;
+      return {Variable(undefined_tensor)};
+    }
+  };
+
+  Variable x(torch::ones(1, torch::requires_grad()));
```

Comment:
@glaringlee I am not sure what are the new naming conventions on the c++ side. But shouldn't these just be `at::Tensor`?

Reply:
I was following other tests as example but I'll try to remove the Variable wrapper.

Reply:
Yes that's what I was expecting. Whichever is best for you works.